torch_xla/csrc/data_ops.cpp

```diff
@@ -100,6 +100,17 @@ xla::XlaOp SqueezeAllTrivialDimensions(xla::XlaOp input) {
   return XlaHelpers::DynamicReshape(input, output_sizes);
 }
 
+xla::XlaOp BuildDynamic(xla::XlaOp input,
+                        absl::Span<const xla::XlaOp> output_sizes) {
+  for (int i = 0; i < output_sizes.size(); i++) {
+    xla::Shape dim_shape = XlaHelpers::ShapeOfXlaOp(output_sizes[i]);
+    if (dim_shape.is_dynamic()) {
```

Comment:
and if the shape isn't dynamic we will be using static dimensions from `InferShape`? I'm asking because we would like shapes to appear symbolically in a graph and I'm not sure this would happen if dimensions are static? @miladm @JackCaoG

Reply:
Yes, dimensions from `InferShape` set shape. This means when a dimension is static, the `upper_bound` values are expected to be equal to the "true" dimensions. Does this align with your side @Krovatkin?

Can you elaborate on: `we would like shapes to appear symbolically in a graph and I'm not sure this would happen if dimensions are static`?